code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,495 @@
+
+/obj/effect/abstract/cheese_trap/Crossed(atom/movable/AM, oldloc)
+	. = ..()
+	if(isrobot(AM) || ishuman(AM))
+		var/turf/T = get_turf(src)
+		for(var/obj/structure/telecomms_doomsday_device/DD in urange(15, T))
+			if(DD.thief)
+				DD.start_the_party(TRUE)
+				return
+
+/obj/machinery/autolathe/trapped
+	name = "recharger"
+	desc = "A charging dock for energy based weaponry. Did it just-"
+	icon_state = "autolathe_trap"
+	//Has someone put an item in the autolathe, breaking the hologram?
+	var/disguise_broken = FALSE
+
+/obj/machinery/autolathe/trapped/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, PROC_REF(material_container_shenanigins))
+
+/obj/machinery/autolathe/trapped/proc/material_container_shenanigins(datum/source, obj/item/attacker, mob/user)
+	if(!disguise_broken)
+		to_chat(user, "<span class='danger'>As you stick [attacker] into the recharger, it sparks and flashes blue. Wait a minute, this isn't a recharger!</span>")
+		name = "modified autolathe"
+		desc = "An autolathe modified with holopad parts, to make it look like a harmless weapon recharger!"
+		do_sparks(3, 1, src)
+		icon_state = "autolathe"
+		disguise_broken = TRUE
+
+/obj/machinery/shieldgen/telecomms
+	name = "overclocked shield generator"
+	desc = "These shield generators seem to have been rewired a lot."
+	anchored = TRUE
+	shield_range = 6
+	req_access = list(ACCESS_ENGINE)
+
+/obj/machinery/shieldwallgen/telecomms
+	icon_state = "Shield_Gen +a" // should avoid misplacing with this
+	anchored = TRUE
+	activated = TRUE
+	req_access = list(ACCESS_TCOMSAT)
+
+/obj/machinery/shieldwallgen/telecomms/Initialize(mapload)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(activate)), 5 MINUTES) // Let the bloody powernet start up, no one will get in this ruin in the first 5 minutes of the map template *initializing*, much less roundstart
+
+/mob/living/silicon/decoy/telecomms
+	faction = list("malf_drone")
+	bubble_icon = "swarmer"
+	name = "D.V.O.R.A.K"
+	desc = "A Downloadable and Versatile, fully Overclocked and Reactive Ai Kernel."
+	icon_state = "ai-triumvirate-malf"
+	universal_speak = TRUE
+	universal_understand = TRUE
+	var/has_died = FALSE // fucking decoy silicons are weird.
+
+/mob/living/silicon/decoy/telecomms/death(gibbed)
+	if(has_died)
+		return ..()
+	has_died = TRUE
+	for(var/obj/structure/telecomms_doomsday_device/D in range(5, src))
+		D.start_the_party()
+		break
+	new /obj/item/documents/syndicate/dvorak_blackbox(get_turf(src))
+	if(prob(50))
+		if(prob(80))
+			new /obj/item/surveillance_upgrade(get_turf(src))
+		else // 10% chance
+			new /obj/item/malf_upgrade(get_turf(src))
+	return ..()
+
+/obj/structure/telecomms_trap_tank
+	name = "rigged plasma tank"
+	desc = "That plasma tank seems rigged to explode!"
+	icon = 'icons/atmos/tank.dmi'
+	icon_state = "toxins_map"
+	anchored = TRUE
+	layer = DISPOSAL_PIPE_LAYER
+	plane = FLOOR_PLANE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
```

Comment:
Could you make this explode on fire_act? Feels like this is basically a red barrel

Reply:
I probably will not because this will gib or at minimum delimb explorers that set off a welding tank located approximately 5 tiles north of one, lmao